hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/protocol/DatanodeDetails.java

```diff
@@ -448,11 +449,15 @@ public HddsProtos.DatanodeDetailsProto getProtoBufMessage() {
   }
 
   public HddsProtos.DatanodeDetailsProto toProto(int clientVersion) {
-    return toProtoBuilder(clientVersion).build();
+    return toProtoBuilder(clientVersion, Collections.emptySet()).build();
+  }
+
+  public HddsProtos.DatanodeDetailsProto toProto(int clientVersion, Set<Port.Name> requiredPorts) {
+    return toProtoBuilder(clientVersion, requiredPorts).build();
   }
 
   public HddsProtos.DatanodeDetailsProto.Builder toProtoBuilder(
-      int clientVersion) {
+      int clientVersion, Set<Port.Name> requiredPorts) {
```

Comment:
Nit: Let's add javadocs here and specify that empty set means that all ports will be included.

Reply:
`requiredPorts` does tend to be misleading I have changed it to `filterPorts`